### DISCUSSION TITLE
Materialising Across Different Partition Dimensions

### DISCUSSION THREAD

**Discussion by bmarcj** [dagster-io/dagster#18025] [Q&A] [answered]
The situation:

I have a series of software defined assets with straightforward dependencies:

A -> B

These are daily time partitioned. There is a scheduled job that updates each asset every day.

The complication:

It turns out that occasionally, parts of the set of data in A are stale and need to be re-materialised across a different dimension ('symbol'), for all days. This changes A to be multi-partitioned across [day, symbol], where symbol is a (dynamic) partition. Which 'symbol' partitions need to be re-materialised is based on contents of a new asset X, which is daily materialised. The dependencies now look like this:

X -> A -> B 

How do I write a schedule that materialises each daily partition at the end of each day, but _before materialising B_ also re-materialises any relevant 'symbol' partitions indicated by X across all time partitions? The condition on materialising those extra partitions before materialising B is why I haven't looked at Sensors or similar event driven approaches. I'd also prefer to keep it simple.

(At this point, it seems like the simplest solution is to manually handle the extra partition dimension in the definition of A i.e. leave dagster unaware of the extra partition, and simply handle the additional 'symbol' partition materialisation manually before progressing on with the usual daily partitioning. This obviously isn't ideal; I'd rather there was an easy 'dagster' way to do this.)

**Comment by sryza** [comment, marked as answer]
@bmarcj this isn't a full answer to your question, but here's some content on how to write a schedule that requests multiple partitions of an asset on every tick: https://github.com/dagster-io/dagster/discussions/15532

**Comment by bmarcj** [comment]
Thanks @sryza. I'm going to return to this at some point when I have a bit of spare time; for now I've opted just to handle I/O externally to dagster's IO framework.
